application/models/File.php

```diff
@@ -154,6 +161,20 @@ public function getProperty($property)
         }
     }
 
+    public function getAltText() {
+        $fileCustomAltText = $this->alt_text;
+        $fileAltTextElement = get_option('file_alt_text_element');
+
+        if ($fileCustomAltText) {
+            return $fileCustomAltText;
+        } elseif ($fileAltTextElement) {
+            $fileAltTextElementText = metadata($this, explode(',',$fileAltTextElement));
+            return strip_formatting($fileAltTextElementText);
```

Comment:
In general for all this, especially if you change the option to use get_table_options, you might take inspiration from the ElementText mixin's getDisplayTitle. It has some fluff you won't need (like it allows for some HTML tags) but the basic structure of getting the text, checking for the html flag, stripping, decoding, is probably what you'd want/need to properly support possible HTML values here.

If you're in here with an element ID only as the option value, you can use getElementById and getElementTextsByRecord to get the text record you'd need.